### PR TITLE
fix: invalid profile case

### LINKED ISSUE #7751: [UI bug] [Export as skill] Confirm skill endpoints is blank when publish profile is imported from existing
Export as a skill->Create a new publish profile-> import from existing

Issue 1:
The confirm skill end point UI turns up blank, 
image

Issue 2 : 
Back button should take the user back to the publishing step but instead it takes the user back to the allowed caller selection

### PR DESCRIPTION
**Pull request by alanlong9278**
## Description

fix invalid profile err in skill experience.
 Current target is invalid,
- case 1: There are valid targets in publish targets.
selectProfile1

- case 2: The whole publish targets are invalid.
invalid

## Task Item
closes #7751

### CODE REVIEW

**Comment by luhan2017** [comment]
The second scenario, instead of two redirection, can we go to the create new publish profile directly from the confirm endpoint page?

File: Composer/packages/client/src/pages/design/exportSkillModal/content/SelectProfile.tsx [hunk 1]
```diff
@@ -24,6 +27,16 @@ const styles = {
   `,
 };
 
+const actionButton = {
+  root: {
+    fontSize: 14,
+    fontWeight: FontWeights.regular,
+    color: SharedColors.cyanBlue10,
```
Comment (hatpick) [inline Composer/packages/client/src/pages/design/exportSkillModal/content/SelectProfile.tsx:34]: why are you overwriting default color of the ActionButton?

File: Composer/packages/client/src/pages/design/exportSkillModal/content/SelectProfile.tsx [hunk 2]
```diff
@@ -76,6 +108,24 @@ export const getManifestId = (
   return fileId;
 };
 
+const onRenderTitle = (options: IDropdownOption[] | undefined): JSX.Element | null => {
+  const option = options?.[0];
+
+  return option ? (
+    <div style={{ display: 'flex', alignItems: 'center' }}>
+      <span>{option.text}</span>
+      {option.data && option.data.icon && (
```
Comment (hatpick) [inline Composer/packages/client/src/pages/design/exportSkillModal/content/SelectProfile.tsx:117]: can be one condition: ```option.data?.icon```

File: Composer/packages/client/src/pages/design/exportSkillModal/content/SelectProfile.tsx [hunk 3]
```diff
@@ -173,38 +244,49 @@ export const SelectProfile: React.FC<ContentProps> = ({
...
+            placeholder={formatMessage('The endpoint url of your web app resource')}
+            styles={{ root: { paddingBottom: '8px' } }}
+            value={endpointUrl}
+            onRenderLabel={onRenderLabel}
+          />
+          <TextField
+            disabled
+            required
+            ariaLabel={formatMessage('The app id of your application registration')}
+            label={formatMessage('Microsoft App Id')}
```
Comment (hatpick) [inline Composer/packages/client/src/pages/design/exportSkillModal/content/SelectProfile.tsx:282]: App ID

File: Composer/packages/client/src/pages/design/exportSkillModal/content/SelectProfile.tsx [hunk 4]
```diff
@@ -173,38 +244,49 @@ export const SelectProfile: React.FC<ContentProps> = ({
...
+            placeholder={formatMessage('The app id')}
```
Comment (hatpick) [inline Composer/packages/client/src/pages/design/exportSkillModal/content/SelectProfile.tsx:283]: can you consult UX for a better placeholder?

File: Composer/packages/client/src/pages/design/exportSkillModal/content/SelectProfile.tsx [hunk 5]
```diff
@@ -55,6 +68,25 @@ const onRenderLabel = (props) => {
   );
 };
 
+const onRenderInvalidProfileWarning = (hasValidProfile, handleShowCreateProfileDialog) => {
+  return (
+    <div>
+      <span>{formatMessage('Publish profile is missing App ID and host name. ')}</span>
```
Comment (hatpick) [inline Composer/packages/client/src/pages/design/exportSkillModal/content/SelectProfile.tsx:74]: shouldn't this be in a warning color? consult UX please

File: Composer/packages/client/src/pages/design/exportSkillModal/content/SelectProfile.tsx [hunk 6]
```diff
@@ -127,43 +180,61 @@ export const SelectProfile: React.FC<ContentProps> = ({
...
+        config.hostname.length > 0
+      );
     } catch (err) {
       console.log(err.message);
       return false;
     }
+  };
+
+  const hasValidProfile = useMemo(() => {
+    if (!publishingTargets) return false;
```
Comment (hatpick) [inline Composer/packages/client/src/pages/design/exportSkillModal/content/SelectProfile.tsx:208]: entire memo can be: ```!!publishingTargets?.some(isValidProfile)```

File: Composer/packages/client/src/pages/design/exportSkillModal/content/SelectProfile.tsx [hunk 7]
```diff
@@ -127,43 +180,61 @@ export const SelectProfile: React.FC<ContentProps> = ({
...
+    return filteredProfile.length > 0;
   }, [publishingTargets]);
 
   const publishingOptions = useMemo(() => {
-    return publishingTargets.map((t) => ({
-      key: t.name,
-      text: t.name,
-    }));
+    return publishingTargets.map((t) =>
+      isValidProfile(t)
```
Comment (hatpick) [inline Composer/packages/client/src/pages/design/exportSkillModal/content/SelectProfile.tsx:217]: move this condition to data:
```data: !isValidProfile(t) ? { ... } : undefined```

File: Composer/packages/client/src/pages/design/exportSkillModal/content/SelectProfile.tsx [hunk 8]
```diff
@@ -127,43 +180,61 @@ export const SelectProfile: React.FC<ContentProps> = ({
...
+            text: t.name,
+            data: { icon: 'TriangleSolid', color: NeutralColors.gray60 },
+          }
+    );
   }, [publishingTargets]);
 
   useEffect(() => {
     setPublishingTargets(settings.publishTargets || []);
     setCurrentTarget((settings.publishTargets || [])[0]);
+    if (!settings.publishTargets || settings.publishTargets.length === 0) {
```
Comment (hatpick) [inline Composer/packages/client/src/pages/design/exportSkillModal/content/SelectProfile.tsx:233]: ```setShowCreationProfileDialog(!settings.publishTargets || settings.publishTargets.length === 0)```

File: Composer/packages/client/src/pages/design/exportSkillModal/content/SelectProfile.tsx (hunk 5 above)
Comment (hatpick) [reply Composer/packages/client/src/pages/design/exportSkillModal/content/SelectProfile.tsx:74]: Ask them about using something like:
https://developer.microsoft.com/en-us/fluentui#/controls/web/messagebar (warning)

**Review by hatpick — CHANGES_REQUESTED** [review]

File: Composer/packages/client/src/pages/design/exportSkillModal/content/SelectProfile.tsx (hunk 2 above)
Comment (alanlong9278) [reply Composer/packages/client/src/pages/design/exportSkillModal/content/SelectProfile.tsx:117]: Fixed.

File: Composer/packages/client/src/pages/design/exportSkillModal/content/SelectProfile.tsx (hunk 4 above)
Comment (alanlong9278) [reply Composer/packages/client/src/pages/design/exportSkillModal/content/SelectProfile.tsx:283]: @vivekkshankar  Could u give me a better placeholder? thanks

File: Composer/packages/client/src/pages/design/exportSkillModal/content/SelectProfile.tsx (hunk 5 above)
Comment (alanlong9278) [reply Composer/packages/client/src/pages/design/exportSkillModal/content/SelectProfile.tsx:74]: Fine, I will replace it.

File: Composer/packages/client/src/pages/design/exportSkillModal/content/SelectProfile.tsx (hunk 6 above)
Comment (alanlong9278) [reply Composer/packages/client/src/pages/design/exportSkillModal/content/SelectProfile.tsx:208]: Refactor, thanks

File: Composer/packages/client/src/pages/design/exportSkillModal/content/SelectProfile.tsx (hunk 8 above)
Comment (alanlong9278) [reply Composer/packages/client/src/pages/design/exportSkillModal/content/SelectProfile.tsx:233]: refactor

File: Composer/packages/client/src/pages/design/exportSkillModal/content/SelectProfile.tsx (hunk 7 above)
Comment (alanlong9278) [reply Composer/packages/client/src/pages/design/exportSkillModal/content/SelectProfile.tsx:217]: refactor

File: Composer/packages/client/src/pages/design/exportSkillModal/content/SelectProfile.tsx (hunk 3 above)
Comment (alanlong9278) [reply Composer/packages/client/src/pages/design/exportSkillModal/content/SelectProfile.tsx:282]: Repalce it

File: Composer/packages/client/src/pages/design/exportSkillModal/content/SelectProfile.tsx (hunk 1 above)
Comment (alanlong9278) [reply Composer/packages/client/src/pages/design/exportSkillModal/content/SelectProfile.tsx:34]: use `messagebar`

**Comment by alanlong9278** [comment]
> The second scenario, instead of two redirection, can we go to the create new publish profile directly from the confirm endpoint page?

There are some check logics before going to publish profile dialog in create publish profile dialog. So I need to merge some logics as common util if we just go to publish profile dialog directly. This may need take some time, so I just give the warning message in this pr and after R13 finished

File: Composer/packages/client/src/pages/design/exportSkillModal/content/SelectProfile.tsx [hunk 9]
```diff
@@ -173,38 +208,49 @@ export const SelectProfile: React.FC<ContentProps> = ({
...
         required
         ariaLabel={formatMessage('Select a publishing profile')}
         label={formatMessage('Publishing Profile')}
         options={publishingOptions}
         placeholder={formatMessage('Select one')}
-        selectedKey={publishingOptions[0].key}
+        selectedKey={selectedKey}
         styles={{ root: { paddingBottom: '8px' } }}
         onChange={handleCurrentProfileChange}
+        onRenderTitle={onRenderTitle}
```
Comment (hatpick) [inline Composer/packages/client/src/pages/design/exportSkillModal/content/SelectProfile.tsx:226]: Q: this is a custom renderer for selected option, but you are always rendering the first item if exist, is that intentional?
https://developer.microsoft.com/en-us/fluentui#/controls/web/dropdown

**Review by hatpick — DISMISSED** [review]

File: Composer/packages/client/src/pages/design/exportSkillModal/content/SelectProfile.tsx (hunk 9 above)
Comment (alanlong9278) [reply Composer/packages/client/src/pages/design/exportSkillModal/content/SelectProfile.tsx:226]: Have checked with the example in fluentui website. Rendering the first item is the default render,  renderer will render the selected option if the selected option changed

Comment (hatpick) [reply Composer/packages/client/src/pages/design/exportSkillModal/content/SelectProfile.tsx:226]: ah, the input arg is the list of selected items and this dropdown is single selection. I get it.

**Comment by alanlong9278** [comment]
@hatpick  Could u pls merge the pr if there is no other problem, thanks.

**Review by hatpick — APPROVED** [review]

**Pull request merged**